packages/web-config-server/src/apiV1/measureData.js

```diff
@@ -330,18 +330,42 @@ export default class extends DataAggregatingRouteHandler {
       entityCode: this.entity.code,
       isDataRegional: dataServices?.[0]?.isDataRegional,
     });
-    dhisApi.injectFetchDataSourceEntities(this.fetchDataSourceEntities);
-    const buildMeasure = async (measureId, ...args) => ({
-      [measureId]: await getMeasureBuilder(measureBuilder)(...args),
-    });
 
+    if (legacy) {
+      const {
+        data_builder: measureBuilder,
+        data_builder_config: measureBuilderConfig,
+      } = mapOverlay;
+      const { dataElementCode = 'value' } = measureBuilderConfig;
```

Comment:
Most other times with this pattern you've had safety around `measureBuilderConfig` being defined - is that not needed here?

Reply:
Usually I'm doing that because I'm protecting against the fact a new style report doesn't have a `measureBuilderConfig`, but this occurrence is within an `if (legacy)` block so we should expect that `measureBuilderConfig` _is_ defined at this point.

Possibly still worth the protection though since technically it's possible for the config to be null in the database

Reply:
No in that case totally leave it unprotected! I'm a fan of non-defensive programming in situations where you have a reasonable expectation, so that if that expectation (slash assumption) isn't met, things blow up. If you were to change anything, I'd make it an assertion that throws a nice error message rather than a defense against it being null. 

See `validateSurveyFields.js` for an example of succinct assertions of preconditions